src/elements/content-sidebar/versions/VersionsSidebarContainer.js

```diff
@@ -121,10 +121,18 @@ class VersionsSidebarContainer extends React.Component<Props, State> {
         });
     };
 
-    handleFetchSuccess = ([fileResponse, versionsResponse]): [BoxItem, FileVersions] => {
+    handleFetchSuccess = ([fileResponse, versionsResponse, currentVersionResponse]): [
+        BoxItem,
+        FileVersions,
+        FileVersions,
+    ] => {
         const { api } = this.props;
         const versionsApi = api.getVersionsAPI(false);
-        const versionsWithCurrent = versionsApi.addCurrentVersion(versionsResponse, fileResponse);
+
+        const versionsWithCurrent: FileVersions = {
+            entries: [...versionsResponse.entries, ...currentVersionResponse.entries],
```

Comment:
Ideally, the munging and decoration logic would live in the API layer and be hidden completely behind `api.getVersionsAPI(false).getVersions()`. I'm not sure how feasible that is given the Feed API limitations, though.

Reply:
agreed this is worth coming back to in a later change.